src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MultipleStringLiteralsCheck.java

```diff
@@ -116,6 +119,17 @@
      */
     public static final String MSG_KEY = "multiple.string.literal";
 
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_KEY_TEXT_BLOCK = "multiple.string.literal.text.block";
```

Comment:
Why new message? For this check, the lines and text blocks must be equal:
```java
String str1 = "foo";
String str2 = """
                     foo
                     """; // violation
```
Now it turns out that the message is different depending on the order of the variables.

Reply:
and there is another case for the method to extract the body of a text block, see https://github.com/checkstyle/checkstyle/pull/8703#discussion_r472006727

Reply:
I made a new message, since unlike the string literal, the content of a text block does not include it's quotation marks.  I didn't feel right about further manipulating the text block content any more than we already are. 

> Why new message? For this check, the lines and text blocks must be equal:
> 
> ```java
> String str1 = "foo";
> String str2 = """
>                      foo
>                      """; // violation
> ```
> 
> Now it turns out that the message is different depending on the order of the variables.

Those two strings are not equal. Only if str2 did not have the trailing whitespace and  additional newline would they be equal. The compiler seems to only remove the first newline, then trim any preceding whitespace for each line, but it is consistent with the line with the least amount of whitespace. Consider:

```
➜  text-blocks-grammar /usr/lib/jvm/java-14-openjdk/bin/javac --enable-preview --source 14 TestClass.java
Note: TestClass.java uses preview language features.
Note: Recompile with -Xlint:preview for details.
➜  text-blocks-grammar cat TestClass.java
public class TestClass {
    public static void main (String... args) {
        String str1 = "foo";
        String str2 = """
                            foo
                """;
        String str3 = """
                foo""";
        String str4 = """
                foo
                """;
        String str5 = """
                      foo
                        foo
                          foo""";

        if (str1.equals(str2)) {
            System.out.println("str1 equals str2");
        }
        if (str1.equals(str3)) {
            System.out.println("str1 equals str3");
        }
        if (str1.equals(str4)) {
            System.out.println("str1 equals str4");
        }
        System.out.println("String 1:");
        System.out.println(str1);
        System.out.println("String 2:");
        System.out.println(str2);
        System.out.println("String 3:");
        System.out.println(str3);
        System.out.println("String 4:");
        System.out.println(str4);
        System.out.println("String 5:");
        System.out.println(str5);


    }
}
➜  text-blocks-grammar /home/nick/.jdks/openjdk-14.0.1/bin/java --enable-preview -classpath /home/nick/Desktop/text-blocks-grammar/out/production/text-blocks-grammar TestClass
str1 equals str3
String 1:
foo
String 2:
            foo

String 3:
foo
String 4:
foo

String 5:
foo
  foo
    foo

```
I agree that we should make a utility method to extract the text block content in this way.  Can I just reference this comment?

Reply:
Sure. I started by trying to combine two of your regular expressions into one. And then realized that a regular expression is not enough and suggested writing a separate method.

Reply:
Added utility in separate commit (https://github.com/checkstyle/checkstyle/pull/8705/commits/9dc02ea7fa3508dee0d7cd857d200f03643c1f13), so that coverage will be 100%.